<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GenericHIDClass</class>
 <widget class="QMainWindow" name="GenericHIDClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>982</width>
    <height>659</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>GenericHID</string>
  </property>
  <property name="windowIcon">
   <iconset resource="generichid.qrc">
    <normaloff>:/GenericHID/ApplicationIcon</normaloff>:/GenericHID/ApplicationIcon</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="margin">
     <number>0</number>
    </property>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tabDesign">
       <attribute name="title">
        <string>Design</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="margin">
         <number>0</number>
        </property>
        <item>
         <widget class="ShapeGraphicsView" name="graphicsView">
          <property name="rubberBandSelectionMode">
           <enum>Qt::IntersectsItemShape</enum>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabTest">
       <attribute name="title">
        <string>Test</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="margin">
         <number>0</number>
        </property>
        <item>
         <widget class="TestWidget" name="testPanel" native="true"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>982</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="separator"/>
    <addaction name="actionSave"/>
    <addaction name="actionSave_As"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuMicrocontroller">
    <property name="title">
     <string>&amp;Microcontroller</string>
    </property>
    <addaction name="actionProgram"/>
    <addaction name="actionExport"/>
    <addaction name="actionImport_and_Program"/>
   </widget>
   <widget class="QMenu" name="menuOptions">
    <property name="title">
     <string>&amp;Options</string>
    </property>
    <addaction name="actionDebug"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuMicrocontroller"/>
   <addaction name="menuOptions"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionPointerTool"/>
   <addaction name="actionWireLinkTool"/>
   <addaction name="actionRotateTool"/>
   <addaction name="actionMirrorTool"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QToolBar" name="toolToolBar">
   <property name="minimumSize">
    <size>
     <width>0</width>
     <height>0</height>
    </size>
   </property>
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="iconSize">
    <size>
     <width>48</width>
     <height>48</height>
    </size>
   </property>
   <attribute name="toolBarArea">
    <enum>LeftToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QDockWidget" name="dockWidget">
   <property name="features">
    <set>QDockWidget::NoDockWidgetFeatures</set>
   </property>
   <property name="allowedAreas">
    <set>Qt::RightDockWidgetArea</set>
   </property>
   <property name="windowTitle">
    <string>Properties</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QVBoxLayout" name="verticalLayout">
     <property name="spacing">
      <number>0</number>
     </property>
     <property name="margin">
      <number>0</number>
     </property>
     <item>
      <widget class="ShapePropertyBrowser" name="listView"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QToolBar" name="toolBarEdit">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionNew"/>
   <addaction name="actionOpen"/>
   <addaction name="actionSave"/>
  </widget>
  <widget class="QDockWidget" name="dockHelp">
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>Help</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <property name="spacing">
      <number>0</number>
     </property>
     <property name="margin">
      <number>0</number>
     </property>
     <item>
      <widget class="QTextBrowser" name="textBrowser"/>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionOpen">
   <property name="icon">
    <iconset resource="generichid.qrc">
     <normaloff>:/GenericHID/FileOpen</normaloff>:/GenericHID/FileOpen</iconset>
   </property>
   <property name="text">
    <string>&amp;Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset resource="generichid.qrc">
     <normaloff>:/GenericHID/FileSave</normaloff>:/GenericHID/FileSave</iconset>
   </property>
   <property name="text">
    <string>&amp;Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSave_As">
   <property name="text">
    <string>Save &amp;As</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+X</string>
   </property>
  </action>
  <action name="actionProgram">
   <property name="text">
    <string>&amp;Program...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+P</string>
   </property>
  </action>
  <action name="actionExport">
   <property name="text">
    <string>&amp;Export</string>
   </property>
  </action>
  <action name="actionImport_and_Program">
   <property name="text">
    <string>&amp;Import and Program</string>
   </property>
  </action>
  <action name="actionPointerTool">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="generichid.qrc">
     <normaloff>:/GenericHID/Cursor</normaloff>:/GenericHID/Cursor</iconset>
   </property>
   <property name="text">
    <string>Pointer</string>
   </property>
  </action>
  <action name="actionWireLinkTool">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="generichid.qrc">
     <normaloff>:/GenericHID/resources/wire.png</normaloff>:/GenericHID/resources/wire.png</iconset>
   </property>
   <property name="text">
    <string>Wire Link</string>
   </property>
  </action>
  <action name="actionRotateTool">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="generichid.qrc">
     <normaloff>:/GenericHID/resources/rotate.png</normaloff>:/GenericHID/resources/rotate.png</iconset>
   </property>
   <property name="text">
    <string>Rotate</string>
   </property>
  </action>
  <action name="actionMirrorTool">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="generichid.qrc">
     <normaloff>:/GenericHID/resources/mirror.png</normaloff>:/GenericHID/resources/mirror.png</iconset>
   </property>
   <property name="text">
    <string>Mirror</string>
   </property>
  </action>
  <action name="actionNew">
   <property name="icon">
    <iconset resource="generichid.qrc">
     <normaloff>:/GenericHID/FileNew</normaloff>:/GenericHID/FileNew</iconset>
   </property>
   <property name="text">
    <string>New</string>
   </property>
  </action>
  <action name="actionDebug">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>&amp;Debug</string>
   </property>
   <property name="toolTip">
    <string>Enable Debug</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>&amp;About</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>ShapePropertyBrowser</class>
   <extends>QListView</extends>
   <header>shapepropertybrowser.h</header>
  </customwidget>
  <customwidget>
   <class>ShapeGraphicsView</class>
   <extends>QGraphicsView</extends>
   <header>shapegraphicsview.h</header>
  </customwidget>
  <customwidget>
   <class>TestWidget</class>
   <extends>QWidget</extends>
   <header>testwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="generichid.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionOpen</sender>
   <signal>triggered()</signal>
   <receiver>GenericHIDClass</receiver>
   <slot>onFileOpen()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave</sender>
   <signal>triggered()</signal>
   <receiver>GenericHIDClass</receiver>
   <slot>onFileSave()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave_As</sender>
   <signal>triggered()</signal>
   <receiver>GenericHIDClass</receiver>
   <slot>onFileSaveAs()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>GenericHIDClass</receiver>
   <slot>onFileExit()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionProgram</sender>
   <signal>triggered()</signal>
   <receiver>GenericHIDClass</receiver>
   <slot>onMicrocontrollerProgram()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionExport</sender>
   <signal>triggered()</signal>
   <receiver>GenericHIDClass</receiver>
   <slot>onMicrocontrollerExport()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionImport_and_Program</sender>
   <signal>triggered()</signal>
   <receiver>GenericHIDClass</receiver>
   <slot>onMicrocontrollerImportAndProgram()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRotateTool</sender>
   <signal>triggered()</signal>
   <receiver>GenericHIDClass</receiver>
   <slot>onRotateTool()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>577</x>
     <y>398</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionMirrorTool</sender>
   <signal>triggered()</signal>
   <receiver>GenericHIDClass</receiver>
   <slot>onMirrorTool()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>577</x>
     <y>398</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionPointerTool</sender>
   <signal>triggered()</signal>
   <receiver>GenericHIDClass</receiver>
   <slot>onPointerTool()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>577</x>
     <y>398</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionWireLinkTool</sender>
   <signal>triggered()</signal>
   <receiver>GenericHIDClass</receiver>
   <slot>onWireLinkTool()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>577</x>
     <y>398</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionNew</sender>
   <signal>triggered()</signal>
   <receiver>GenericHIDClass</receiver>
   <slot>onFileNew()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>490</x>
     <y>334</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAbout</sender>
   <signal>triggered()</signal>
   <receiver>GenericHIDClass</receiver>
   <slot>onHelpAbout()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>490</x>
     <y>329</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionDebug</sender>
   <signal>triggered()</signal>
   <receiver>GenericHIDClass</receiver>
   <slot>onOptionsDebug()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>490</x>
     <y>329</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onFileOpen()</slot>
  <slot>onFileSave()</slot>
  <slot>onFileSaveAs()</slot>
  <slot>onFileExit()</slot>
  <slot>onMicrocontrollerProgram()</slot>
  <slot>onMicrocontrollerExport()</slot>
  <slot>onMicrocontrollerImportAndProgram()</slot>
  <slot>onRotateTool()</slot>
  <slot>onMirrorTool()</slot>
  <slot>onPointerTool()</slot>
  <slot>onWireLinkTool()</slot>
  <slot>onFileNew()</slot>
  <slot>onOptionsDebug()</slot>
  <slot>onHelpAbout()</slot>
 </slots>
</ui>
